src/render-to-layer.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Events from './utils/events';
+import Dom from './utils/dom';
+import debounce from 'lodash.debounce';
+
+// heavily inspired by https://github.com/Khan/react-components/blob/master/js/layered-component-mixin.jsx
+const RenderToLayer = React.createClass({
+
+  componentDidMount() {
+    this._renderLayer();
+  },
+
+  componentDidUpdate() {
+    this._renderLayer();
+  },
+
+  componentWillUnmount() {
+    this._unbindClickAway();
+    if (this._layer) {
+      this._unrenderLayer();
+    }
+  },
+
+  _checkClickAway(e) {
+    debugger
+    if (!this.canClickAway) {
+      return;
+    }
+    const el = this._layer;
+    if (e.target !== el &&
+        !Dom.isDescendant(el, e.target) &&
+        document.documentElement.contains(e.target)) {
+      if (this.props.componentClickAway) {
+        this.props.componentClickAway(e);
+      }
+    }
+  },
+
+  _preventClickAway(e) {
+    if (e.detail === this) {
+      return;
+    }
+    this.canClickAway = false;
+  },
+
+  _allowClickAway() {
+    this.canClickAway = true;
+  },
+
+  getLayer() {
+    return this._layer;
+  },
+  
+  render() {
+    return null;
+  },
+
+  _renderLayer() {
+    debugger
```

Comment:
Should be it now

On Tue 3 Nov 2015 11:41 Olivier Tassinari <notifications@github.com> wrote:

> In src/render-to-layer.js
> <https://github.com/callemall/material-ui/pull/2042#discussion_r43738753>:
>
> > +  },
> > +
> > +  _allowClickAway() {
> > +    this.canClickAway = true;
> > +  },
> > +
> > +  getLayer() {
> > +    return this._layer;
> > +  },
> > +
> > +  render() {
> > +    return null;
> > +  },
> > +
> > +  _renderLayer() {
> > +    debugger
>
> Could you push your changes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/callemall/material-ui/pull/2042/files#r43738753>.
>